moto/core/responses.py

```diff
@@ -115,7 +115,10 @@ def _dispatch(self, request, full_url, headers):
         if not querystring:
             querystring.update(parse_qs(urlparse(full_url).query, keep_blank_values=True))
         if not querystring:
-            querystring.update(parse_qs(self.body, keep_blank_values=True))
+            body = self.body
+            if six.PY3 and isinstance(body, bytes):
```

Comment:
Any reason to do this as opposed to `if isinstance(body, six.binary_type)`? We want to do the decoding in Py2, right?